src/rf/apps/workers/process.py

```diff
@@ -58,48 +63,84 @@ def start_polling(self):
+        def ensure_band_count(key_string, range=None):
+            """
+            Gets the first `range` bytes from the s3 resource and uses it to
+            determine the number of bands in the image.
+            """
+            connection = S3Connection()
+            bucket = connection.get_bucket(settings.AWS_BUCKET_NAME)
+            s3_key = Key(bucket)
+            s3_key.key = key_string
+            random_filename = '/tmp/' + str(uuid.uuid4())
+            tempfile = open(random_filename, 'w')
+            if range is not None:
+                range = {'Range': 'bytes=' + range}
+                s3_key.get_contents_to_file(tempfile, headers=range)
+            else:
+                s3_key.get_contents_to_file(tempfile)
+
+            tempfile.close()
+            try:
+                validator = gdal.Open(random_filename)
+                # Tiler needs 3+ bands.
+                raster_ok = validator.RasterCount >= 3
+            except:
+                raster_ok = False
+
+            os.remove(random_filename)
+            return raster_ok
+
+        def mark_image_valid(s3_uuid, url):
+            image = LayerImage.objects.get(s3_uuid=s3_uuid)
             image.status = enums.STATUS_VALIDATED
             image.save()
             layer_id = image.layer_id
-        except:
-            payload_uuid = '1aa064aa-1086-4ff1-a90b-09d3420e0343'
-            layer_id = 2
 
-        data = {'url': url, 's3_uuid': payload_uuid}
-        payload = self.make_payload(JOB_REPROJECT, data)
-        self.post_to_queue(payload)
+            data = {'url': url, 's3_uuid': s3_uuid}
+            payload = self.make_payload(JOB_REPROJECT, data)
+            self.post_to_queue(payload)
 
-        # Add a message to the queue that we can use to watch for timeouts.
-        data = {
-            'timeout': time.time() + TIMEOUT_SECONDS,
-            'layer_id': layer_id
-        }
-        payload = self.make_payload(JOB_TIMEOUT, data)
-        self.post_to_queue(payload, TIMEOUT_DELAY)
-        return True
+            # Add a message to the queue that we can use to watch for timeouts.
+            data = {
+                'timeout': time.time() + TIMEOUT_SECONDS,
+                'layer_id': layer_id
+            }
+            payload = self.make_payload(JOB_TIMEOUT, data)
+            self.post_to_queue(payload, TIMEOUT_DELAY)
+            return True
+
+        def mark_image_invalid(s3_uuid):
+            image = LayerImage.objects.get(s3_uuid=s3_uuid)
+            image.status = enums.STATUS_INVALID
+            image.save()
+            layer_id = image.layer_id
+            layer = Layer.objects.get(id=layer_id)
+            layer.status = enums.STATUS_FAILED
+            layer.save()
+            return True
+
+        key = data['s3']['object']['key']
+        s3_uuid = self.extract_uuid_from_aws_key(key)
+        url = self.make_s3_url(data['s3']['bucket']['name'], key)
+        range = '0-1000000'  # Get first Mb(ish) of bytes.
```

Comment:
Rob did mention some cases though where he suspected we might need then entire image in order to do a proper verification. I don't know what those cases are or how to reproduce them though.